lib/datadog/tracing/contrib/roda/instrumentation.rb

```diff
@@ -0,0 +1,79 @@
+require_relative '../../metadata/ext'
+require_relative 'ext'
+require_relative '../analytics'
+module Datadog
+  module Tracing
+    module Contrib
+      module Roda
+        # Instrumentation for Roda
+        module Instrumentation
+
+          def _roda_handle_main_route
+            instrument do
+              super
+            rescue => e
+              ['500'] # [status, headers, body]
```

Comment:
After testing this iteration out on an endpoint that should have thrown `undefined local variable or method`, it looks like server errors now throw an `undefined method `each' for nil:NilClass` for all the 5xx type errors that I tried out.

If I comment out `ddtrace`, the web server returns back `undefined local variable or method`.

Any concerns if I revert the rescue block back into the `instrument` method so it continues pass the status/headers/body without being affected by ddtrace? 

https://github.com/DataDog/dd-trace-rb/blob/aa1db9727d2b048865e47f505db306c5872c9ead/lib/datadog/tracing/contrib/roda/instrumentation.rb#L50-L66

Reply:
The new changes are incorrect for error handling: the error is now swallowed by `lib/datadog/tracing/contrib/roda/instrumentation.rb`.

Please revert back to the previous iteration.

Reply:
Thanks! This definitely makes sense, I'll revert it back!